### ISSUE TITLE
Global variables cannot be accesible, while trigggering mocha parallelly

### ISSUE THREAD

**Issue by Dikshita25** [mochajs/mocha#4751]
### Description
Global variables are not accessible within test files, when set mocha runner to trigger tests parallelly

### Steps to Reproduce
1) I'm using Mocha to trigger the tests programmatically and passing the parallel to execute the tests parallelly, below is the code explaining the same:
```
var Mocha = require('mocha');

const triggerMochaTest = callback => {
  const mocha = new Mocha({
    parallel: true,
    global: ['consumerData']
  });

  mocha.addFile('./test/test1.js');
  mocha.addFile('./test/test2.js');

  mocha
    .run(() => {
      callback();
    })
    .on('test', function(test) {
      console.log('Test started: ' + test.title);
    })
    .on('test end', function(test) {
      console.log('Test done: ' + test.title);
    })
    .on('pass', function(test) {
      console.log('Test passed');
    })
    .on('fail', function(test, err) {
      console.log('Test fail');
      console.log(err);
    })
    .on('end', function() {
      console.log('All done');
    });
};

module.exports = triggerMochaTest;

```
2) And have created a test file, wherein I'm trying to access the global variable for assertions, below is the sample code of the test file:
```
const assert = require('assert');

const expectedOutput = require('../sampleData/expectedData/expectedOutput.json');
const consumerOutput = global.consumerData;

describe('Verify the service output', function() {
  it('Verify the name', function() {
    const userInfo = expectedOutput[consumerOutput.key];

    // comparing the expected output with the service output
    assert.deepEqual(userInfo, consumerOutput.output);
  });
});

```

**Expected behavior:** I should be able to access, the global variable, when running the mocha runner parallelly 

**Actual behavior:** I get an error `TypeError: Cannot read property 'key' of undefined` because the consumerOutput is undefined 

**Reproduces how often:** every time

Note: However, if we don't pass the parallel flag to be true, the execution works fine without any error and the consumerOutput can be accessed successfully

### Versions
- The output of `mocha --version` and `node node_modules/.bin/mocha --version`:  ^9.1.1
- The output of `node --version`: v12.18.1
- Your operating system
  - name and version: macOS Big Sur and version is 11.5.2
  - architecture (32 or 64-bit): NA
- Your shell (e.g., bash, zsh, PowerShell, cmd): zsh
- Your browser and version (if running browser tests): NA
- Any third-party Mocha-related modules (and their versions): NA
- Any code transpiler (e.g., TypeScript, CoffeeScript, Babel) being used (and its version): NA

### Additional Information
NOTE: However, if we don't pass the parallel flag to be true, the execution works fine without any error and the consumerOutput can be accessed successfully

**Comment by juergba** [comment]
@Dikshita25 Where are you declaring `global.consumerData`?

**Comment by Dikshita25** [comment]
Hi @juergba, 
I had declared the global.consumerData into a separate file, but even if we try declaring it in the same file we still face the same issue. The below code is for reference.

`var Mocha = require('mocha');

const triggerMochaTest = callback => {
  global.consumerData = { key: '1' };
  const mocha = new Mocha({
    parallel: true,
    global: ['consumerData']
  });

  mocha.addFile('./test/test1.js');
  mocha.addFile('./test/test2.js');

  mocha
    .run(() => {
      callback();
    })
    .on('test', function(test) {
      console.log('Test started: ' + test.title);
    })
    .on('test end', function(test) {
      console.log('Test done: ' + test.title);
    })
    .on('pass', function(test) {
      console.log('Test passed');
    })
    .on('fail', function(test, err) {
      console.log('Test fail');
      console.log(err);
    })
    .on('end', function() {
      console.log('All done');
    });
};

module.exports = triggerMochaTest;`

Thanks

**Comment by juergba** [comment]
```js
// t1.js
global.mochaVar = 'testString';
const { fork } = require('child_process');
fork('./t2.js');

// t2.js
console.log('global variable:', global.mochaVar);           // prints: "global variable: undefined"
```

In parallel mode each test file is executed in a different and independent worker process, each of them having its own global scope. So it's normal Node behaviour that one process doesn't know the global variables of another process.

In your example you define `global.consumerData` in the parent process which is then out of scope in all child-processes.

**Comment by Dikshita25** [comment]
Thanks, @juergba . But is there any way I can pass an input to the test files from the mocha instance?

**Comment by juergba** [comment]
For little input data you can use [environment variables](https://nodejs.org/docs/latest-v14.x/api/process.html#process_process_env), which are passed from parent to child process. Which is the `process.env` object and your value should be string or number or boolean.

For larger data neither the `global` nor the `process.env` way makes sense. Then put your expected data directly into your test file or store /read it from a json file.

**Comment by Dikshita25** [comment]
Ok thanks for the suggestions @juergba
